c++/src/capnp/layout.c++

```diff
@@ -1777,6 +1836,11 @@ struct WireHelpers {
         }
 
 #if !CAPNP_LITE
+        if (canonical) {
+          KJ_FAIL_REQUIRE("Cannot create a canonical message with a capability") {
+            goto useDefault;
```

Comment:
I think this should probably be `break;` -- when exceptions are disabled we go on executing the code and simply produce a non-canonical message.